pkg/controller/installation/validation.go

```diff
@@ -361,6 +362,22 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 		}
 	}
 
+	// Verify that daemonset affinity ensures that they are not run on virtual nodes (AKS) or fargate nodes (EKS).
+	if instance.Spec.DaemonSetAffinity != nil {
```

Comment:
previously when we've provided an option that would allow removing a defaulted configuration, we do not validate that the configuration retains the defaulted value.
In other words, I don't think we need the validation for either AKS or EKS.

Reply:
@caseydavenport WDYT?

Reply:
There's definitely no rule we need to validate that the config matches the previous defaults. But if we _know_ there are config options here that won't work on certain types of clusters, I think it's fair to validate against that.

That said, the virtual nodes can be avoided using other means than an affinity, so it is probably valid for this field to _not_ include those. This validation might be a tiny bit too strict.

Reply:
There's definitely no rule we need to validate that the config matches the previous defaults. But if we _know_ there are config options here that won't work on certain types of clusters, I think it's fair to validate against that.

That said, the virtual nodes can be avoided using other means than an affinity, so it is probably valid for this field to _not_ include those. This validation might be a tiny bit too strict.